pkg/movingaverage/weight_moving_average.go

```diff
@@ -38,39 +40,36 @@ func NewWMA(sizes ...int) *WMA {
 
 // Add adds a data point.
 func (w *WMA) Add(n float64) {
+	w.score = w.score - w.sum + n*float64(w.size)
+	if w.count < w.size {
+		w.sum = w.sum + n
+	} else {
+		w.sum = w.sum - w.records[(w.count)%w.size] + n
+	}
 	w.records[w.count%w.size] = n
 	w.count++
 }
 
 // Get returns the weight average of the data set.
 func (w *WMA) Get() float64 {
 	if w.count == 0 {
-		return 0
+		return w.score / float64(w.size)
 	}
-
-	sum := 0.0
-
 	if w.count < w.size {
-		for i := 0; i < int(w.count); i++ {
-			sum += w.records[i]
-		}
-		return sum / float64(w.count)
+		return w.score / float64((2.0*w.size-w.count+1)*w.count/2.0)
```

Comment:
2.0*w.size-w.count+1 ?

Reply:
when w is full , the count=w.size+(w.size-w.count+1),eg 
w.size=5 , the w  has 1,1,2,3 the count=4
the count should be  5+4+3+2=14 
it should be same wiht (w.size+(w.size-w.count+1))*w.count/2=14, equal (the first element+last element)*count/2

Reply:
I think it will be better to add some comment in code or replace it with `(w.size+(w.size-w.count+1))*w.count/2`